src/audio/rtnr/rtnr.c

```diff
@@ -530,66 +535,82 @@ static int rtnr_copy(struct comp_dev *dev)
-		source_bytes = frames * audio_stream_frame_bytes(&source->stream);
-		sink_bytes = frames * audio_stream_frame_bytes(&sink->stream);
+		comp_dbg(dev, "rtnr_copy() source->id: %d, frames = %d", source->id, frames);
 
-		buffer_stream_invalidate(source, source_bytes);
+		if (frames) {
+			source_bytes = frames * audio_stream_frame_bytes(&source->stream);
+			sink_bytes = frames * audio_stream_frame_bytes(&sink->stream);
 
-		/* Run processing function */
+			buffer_stream_invalidate(source, source_bytes);
 
-		/*
-		 * Processing function uses an array of pointers to source streams
-		 * as parameter.
-		 */
+			/* Run processing function */
 
-		/* copy required data from sof audio stream to RTNR audio stream */
-		rtnr_copy_from_sof_stream(sources_stream[0], &source->stream);
-		rtnr_copy_from_sof_stream(sink_stream, &sink->stream);
+			/* copy required data from sof audio stream to RTNR audio stream */
+			rtnr_copy_from_sof_stream(sources_stream[0], &source->stream);
+			rtnr_copy_from_sof_stream(sink_stream, &sink->stream);
 
-		cd->rtnr_func(dev, (struct audio_stream_rtnr **)&sources_stream, sink_stream,
-					frames);
+			/*
+			 * Processing function uses an array of pointers to source streams
+			 * as parameter.
+			 */
+			cd->rtnr_func(dev, (struct audio_stream_rtnr **)&sources_stream,
+						sink_stream, frames);
 
-		/*
-		 * real process function of rtnr, consume/produce data from internal queue
-		 * instead of component buffer
-		 */
-		RTKMA_API_Process(cd->rtk_agl, 0, cd->source_rate, MicNum);
+			/*
+			 * real process function of rtnr, consume/produce data from internal queue
+			 * instead of component buffer
+			 */
+			RTKMA_API_Process(cd->rtk_agl, 0, cd->source_rate, MicNum);
 
-		/* copy required data from RTNR audio stream to sof audio stream */
-		rtnr_copy_to_sof_stream(&source->stream, sources_stream[0]);
-		rtnr_copy_to_sof_stream(&sink->stream, sink_stream);
+			/* copy required data from RTNR audio stream to sof audio stream */
+			rtnr_copy_to_sof_stream(&source->stream, sources_stream[0]);
+			rtnr_copy_to_sof_stream(&sink->stream, sink_stream);
 
-		buffer_stream_writeback(sink, sink_bytes);
+			buffer_stream_writeback(sink, sink_bytes);
 
-		/* Track consume and produce */
-		comp_update_buffer_consume(source, source_bytes);
-		comp_update_buffer_produce(sink, sink_bytes);
-	}
+			/* Track consume and produce */
+			comp_update_buffer_consume(source, source_bytes);
+			comp_update_buffer_produce(sink, sink_bytes);
+		}
+	} else {
+		comp_dbg(dev, "rtnr_copy() passthrough");
+
+		/* Get source, sink, number of frames etc. to process. */
+		comp_get_copy_limits_with_lock(source, sink, &cl);
 
+		buffer_stream_invalidate(source, cl.source_bytes);
+
+		audio_stream_copy(&source->stream, 0, &sink->stream, 0,
+				source->stream.channels * cl.frames);
+
+		buffer_stream_writeback(sink, cl.sink_bytes);
+		comp_update_buffer_consume(source, cl.source_bytes);
+		comp_update_buffer_produce(sink, cl.sink_bytes);
```

Comment:
you could do
```
if (!cd->process_enable) {
	...
	return 0;
}
```
in the beginning and then leave the rest as it was or with very few changes, the delta would be much smaller then